### DISCUSSION TITLE
How to package a non-main branch for testing?

### DISCUSSION THREAD

**Discussion by aprilrd** [graphql/graphql-js#3323] [Q&A] [answered]
Hello, we have [a fork to add non-null operators](https://github.com/graphql/graphql-js/pull/3281) to GraphQL. We want to integration-test our fork with other libraries, but we don't know how to package the fork. It looks like the repo relies on a custom script on Github Action to package it. 

What would be the best way to package this branch?

**Comment by saihaj** [comment]
similar to [`defer-and-stream`](https://github.com/graphql/graphql-js#experimental-features) we can add this as an experimental feature. cc @IvanGoncharov

**Comment by IvanGoncharov** [comment, marked as answer]
> It looks like the repo relies on a custom script on Github Action to package it.

@aprilrd Not really you can do this locally just do:
```
npm install
npm run build:npm
```
You will get `npmDist` folder with `package.json` in it.
To pack it as a package you can do `npm pack ./npmDist`.

If you want to do something more official we can use the same approach as `stream-defer` that @saihaj mentioned.

**Reply by aprilrd** [reply]
Thank you for the pointer! I will give that a go and mark this as an answer if it works for me.

**Reply by aprilrd** [reply]
Your guidance worked beautifully. Thank you!
